### ISSUE TITLE
[N300] TTNN Unit Test Failures: AttributeError: 'NoneType' object has no attribute 'enable_program_cache'

### ISSUE THREAD

**Issue by cfjchu** [tenstorrent-metal/tt-metal#6989]
```
tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=16-input_channels=3-input_height=1056-input_width=160-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override={'act_block_h': 160}-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=16-input_channels=16-input_height=1056-input_width=160-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override={'act_block_h': 160}-use_shallow_conv_variant=False0] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=16-input_channels=16-input_height=528-input_width=80-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False0] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=32-input_channels=16-input_height=264-input_width=40-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=32-input_channels=32-input_height=264-input_width=40-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False0] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=32-input_channels=32-input_height=132-input_width=20-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False0] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=64-input_channels=32-input_height=66-input_width=10-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=64-input_channels=64-input_height=66-input_width=10-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=32-input_channels=96-input_height=132-input_width=20-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=32-input_channels=32-input_height=132-input_width=20-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False1] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=32-input_channels=64-input_height=264-input_width=40-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=32-input_channels=32-input_height=264-input_width=40-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False1] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=16-input_channels=48-input_height=528-input_width=80-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override={'act_block_h': 32}-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=16-input_channels=16-input_height=528-input_width=80-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False1] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=16-input_channels=32-input_height=1056-input_width=160-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override={'act_block_h': 160}-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=16-input_channels=16-input_height=1056-input_width=160-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override={'act_block_h': 160}-use_shallow_conv_variant=False1] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT8_B-batch_size=2-output_channels=1-input_channels=16-input_height=1056-input_width=160-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override={'act_block_h': 160}-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=16-input_channels=3-input_height=1056-input_width=160-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override={'act_block_h': 160}-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=16-input_channels=16-input_height=1056-input_width=160-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override={'act_block_h': 160}-use_shallow_conv_variant=False0] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=16-input_channels=16-input_height=528-input_width=80-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False0] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=32-input_channels=16-input_height=264-input_width=40-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=32-input_channels=32-input_height=264-input_width=40-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False0] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=32-input_channels=32-input_height=132-input_width=20-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False0] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=64-input_channels=32-input_height=66-input_width=10-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=64-input_channels=64-input_height=66-input_width=10-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=32-input_channels=96-input_height=132-input_width=20-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=32-input_channels=32-input_height=132-input_width=20-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False1] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=32-input_channels=64-input_height=264-input_width=40-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=32-input_channels=32-input_height=264-input_width=40-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False1] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=16-input_channels=48-input_height=528-input_width=80-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override={'act_block_h': 32}-use_shallow_conv_variant=False] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
ERROR tests/ttnn/unit_tests/operations/test_conv2d.py::test_unet_conv_wh[output_layout=Layout.ROW_MAJOR-math_fidelity=MathFidelity.LoFi-activations_dtype=DataType.BFLOAT8_B-weights_dtype=DataType.BFLOAT16-batch_size=2-output_channels=16-input_channels=16-input_height=528-input_width=80-filter_height=3-filter_width=3-stride_h=1-stride_w=1-pad_h=1-pad_w=1-use_1d_systolic_array=True-config_override=None-use_shallow_conv_variant=False1] - AttributeError: 'NoneType' object has no attribute 'enable_program_cache'
```

fyi @xanderchin  @jliangTT

**Comment by cfjchu** [comment]
@nsmithtt we're trying to enable ttnn unit tests in CI.. we see a lot of failing `test_unet_conv_wh`. The plan is the enable all current passing unit tests and disable the failing ones. Since you're on unet, can you or someone take a look at resolving these issues? This failure happens on N150 and N300.

**Comment by nsmithtt** [comment]
@tapspatel, can you take a look?

**Comment by tapspatel** [comment]
on it

**Comment by cfjchu** [comment]
@tapspatel @AleksKnezevic @mtatsumiTT FYI you can run a reduced CI regression ttnn test-suite across GS/N150/N300 here when you test:
https://github.com/tenstorrent-metal/tt-metal/actions/workflows/ttnn-post-commit-wrapper.yaml

**Comment by tapspatel** [comment]
Issue is there needs to be a strict ordering of parameters for a pytest. "Device" need to appear first in the pytest function. Changing the ordering for the affected functions resolves the issue. Putting up a PR

**Comment by tapspatel** [comment]
=========== 249 passed, 665 skipped, 5 warnings in 333.85s (0:05:33) ===========

Runs till completion

**Comment by tapspatel** [comment]
PR: https://github.com/tenstorrent-metal/tt-metal/pull/7080

**Comment by cfjchu** [comment]
awesome, thanks @tapspatel

**Comment by tapspatel** [comment]
Another side note, for these unit tests, for n300, we need to put the dispatch onto the eth cores, otherwise the smaller core grid size will fall divisor computation. Ill fix these tests to be more dynamic, although this will be an issue across maxpool as well (and any other op that depends on specific sharded specs)

**Comment by tapspatel** [comment]
Merged: 93d19bd4f1c863e873ee6f673c516f4158a3763e
